src/rabbit_registry.erl

```diff
@@ -114,6 +116,29 @@ conditional_unregister({exchange_decorator, Type}) ->
 conditional_unregister(_) ->
     ok.
 
+% update channels to enable/disable interceptors
+registered({{Class, Type}, ModuleName}) ->
+    ClassModule = class_module(Class),
+    case proplists:get_value(registered, ClassModule:module_info(functions)) of
```

Comment:
Agree that a behaviour would be helpful. Not sure what function names should be there. Maybe straightforward `added_to_registry`, `removed_from_registry`?